internal/util/debug/debug.go

```diff
@@ -123,3 +142,80 @@ func RunHandler(ctx context.Context, addr string, r prometheus.Registerer, l *za
 	s.Close()
 	l.Sugar().Info("Debug server stopped.")
 }
+
+func getArchiveHandler() http.HandlerFunc {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		var (
+			scheme          string
+			u               url.URL
+			responses       = make(map[string][]byte, 2)
+			debugFilePrefix = "FerretDB"
+			debugFileName   = "debug.zip"
+			err             error
+		)
+
+		u.Path = metricsPath
+		u.Host = req.Host
+
+		if req.URL.Scheme == "" {
+			scheme = "http"
+		}
+
+		u.Scheme = scheme
+
+		metricsFileName := generateFileName(debugFilePrefix, filepath.Base(metricsPath)+".txt")
+		responses[metricsFileName], err = performRequest(u)
+		if err != nil {
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		u.Path = pprofPath
+		pprofFileName := generateFileName(debugFilePrefix, filepath.Base(pprofPath)+".html")
+		responses[pprofFileName], err = performRequest(u)
+		if err != nil {
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		rw.Header().Set("Content-Type", "application/zip")
+		rw.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=%s", generateFileName(debugFilePrefix, debugFileName)))
+
+		zipWriter := zip.NewWriter(rw)
+		defer zipWriter.Close()
+
+		for fileName, response := range responses {
+			fileWriter, err := zipWriter.Create(fileName)
+			if err != nil {
+				http.Error(rw, fmt.Sprintf("fail creating file %s , (%s)", fileName, err.Error()), http.StatusInternalServerError)
+				return
+			}
+			_, err = io.Copy(fileWriter, bytes.NewReader(response))
+			if err != nil {
+				http.Error(rw, fmt.Sprintf("failed - adding %s to zip , (%s)", fileName, err.Error()), http.StatusInternalServerError)
```

Comment:
I think we should log error like that, but don't return them

Reply:
we can do that. But since this is specifically for debugging , isn't it good to return an error , so for any issues , error is clearly known. it shall not be lost in the logs. WDYT?

Reply:
@AlekSi , WDYT?